core/src/tpu.rs

```diff
@@ -164,7 +164,7 @@ impl Tpu {
         let (_, tpu_quic_t) = spawn_server(
             transactions_quic_sockets,
             keypair,
-            cluster_info.my_contact_info().tpu.ip(),
+            cluster_info.my_contact_info().tpu_quic().unwrap().ip(),
```

Comment:
this one

Reply:
> This intentionally stops the operator from **starting** a node with tpu thread but invalid `tpu` address. It can only fail very early on _during the startup phase_ if there is a way to configure the ports incorrectly and still make it to here.

can we at least do it with a helpful message instead of throwing a scary stacktrace at them?  99/100 are going to come to discord with a screenshot and "why?"

Reply:
I changed this one and the ones below to use `expect` indicating which port is invalid, e.g. tpu or gossip etc.
The remaining context is included in the `Error`:
https://github.com/solana-labs/solana/blob/be8e463a5/gossip/src/contact_info.rs#L38-L62
e.g if the port is invalid, etc. and those would be printed in the panic message.